#From the fasta file separate the 16S and the resistance gene part of the sequences.
    #Load the data into python(open or with biopython load as sequence object)
    #Either slice the sequence according to the known sequence or at the expected length (82)
    #Make a new file with the 2 parts of the sequence separated (seq_name[NDM,16S]).
#(eliminate the primer sequences)
#create OTUs (unix)
    #Create clusters based on similarity 
    #Find consensus seq for each cluster
#Blast the consensus seq of each OTU to a 16S database (silva and/or NCBI)
#Create a phylogenetic tree
